scripts/create-references.js

```diff
@@ -11,55 +10,42 @@ const config = JSON.parse(fs.readFileSync('tsconfig.json').toString());
 config.files = [];
 config.references = [];
 
-(async function() {
-
-    const { stdout, stderr } = await exec('yarn workspaces info --json');
-    
-    const lines = stdout.split('\n');
-    const depthTree = lines.slice(1, lines.length - 2).join('\n');
-    let workspaces = undefined;
-
-    // sometimes yarn prints this line: yarn run v1.22.10 and sometimes not
-    try {
-        workspaces = JSON.parse(depthTree);
-    } catch (e) {
-        workspaces = JSON.parse(stdout);
-    }
-
-    for (const name in workspaces) {
-        const workspace = workspaces[name];
-        const location = path.resolve(process.cwd(), workspace.location);
-        const tsconfigPath = path.resolve(location, 'tsconfig.json');
-
-        if (fs.existsSync(tsconfigPath)) {
-            config.references.push({
-                path: workspace.location,
-            });
-            const workspaceConfig = JSON.parse(
-                fs.readFileSync(tsconfigPath).toString(),
-            );
-            workspaceConfig.compilerOptions.composite = true;
-            workspaceConfig.references = [];
-            for (const dependency of workspace.workspaceDependencies) {
-                const dependecyLocation = path.resolve(
-                    process.cwd(),
-                    workspaces[dependency].location,
-                );
-                if (
-                    fs.existsSync(
-                        path.resolve(dependecyLocation, 'tsconfig.json'),
-                    )
-                ) {
-                    workspaceConfig.references.push({
-                        path: path.relative(location, dependecyLocation),
-                    });
-                }
-            }
-            fs.writeFileSync(
-                tsconfigPath,
-                JSON.stringify(workspaceConfig, undefined, 4),
-            );
+(async function () {
+  const { stdout, stderr } = await exec('yarn workspaces info --json');
+
+  const lines = stdout.split('\n');
+  const depthTree = lines.slice(1, lines.length - 2).join('\n');
+  let workspaces = undefined;
+
+  // sometimes yarn prints this line: yarn run v1.22.10 and sometimes not
+  try {
+    workspaces = JSON.parse(depthTree);
+  } catch (e) {
+    workspaces = JSON.parse(stdout);
+  }
+
+  for (const name in workspaces) {
+    const workspace = workspaces[name];
+    const location = path.resolve(process.cwd(), workspace.location);
+    const tsconfigPath = path.resolve(location, 'tsconfig.json');
```

Comment:
I would put these two variables in a `try-catch` block, as resolving paths is sensitive to errors. The path might differ on the server, builds might go wrong or (since this is JavaScript) the `path.resolve` function might just decide it's having a bad day.

In any case, you want to be able to either fallback to a back-up method, refresh the page to try again or let the user know something went wrong. This looks like a function that reaches quite deep into the inner workings of the PWA, so you want to be sure any exceptions are properly handled.

Reply:
`path.resolve` never throws an error, it just concatenates the path. Whether or not it exists. With `fs.existsSync(tsconfigPath)` I check if it actually exists.

This is also a developer script to automatically update the references in `tsconfig.json`, based on the dependencies of the component. The script is not really important.